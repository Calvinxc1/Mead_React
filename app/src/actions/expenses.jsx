import uuid from 'uuid';

const addExpense = ({
    description='',
    notes='',
    amount=0,
    createdAt=0,
  }={}) => ({
  type: 'ADD_EXPENSE',
  expense: {
    id: uuid(),
    description,
    notes,
    amount,
    createdAt,
  }
});

const removeExpense = (id) => ({type: 'REMOVE_EXPENSE', id});

const editExpense = (id, updates) => ({
  type: 'EDIT_EXPENSE',
  id, updates,
});

export {addExpense, removeExpense, editExpense};
