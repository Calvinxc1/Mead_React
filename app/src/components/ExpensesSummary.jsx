import React from 'react';
import {connect} from 'react-redux';
import numeral from 'numeral';

import selectExpenses from './../selectors/expenses.jsx';
import {selectExpensesTotal} from './../selectors/expenses-total.js';

const ExpensesSummary = (props) => <div>
  Viewing {
    props.expenses.length
  } expenses totalling {
    numeral(selectExpensesTotal(props.expenses)/100).format('$0,0.00')
  }
</div>;

const mapStateToProps = (state) => ({
  expenses: selectExpenses(state.expenses, state.filters),
});

export default connect(mapStateToProps)(ExpensesSummary);
export {ExpensesSummary};
