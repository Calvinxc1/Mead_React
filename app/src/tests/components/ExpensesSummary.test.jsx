import React from 'react';
import {shallow} from 'enzyme';

import {ExpensesSummary} from './../../components/ExpensesSummary.jsx';
import expenses from './../fixtures/expenses.jsx';

test('should render ExpensesSummary with multiple expense items', () => {
  const wrapper = shallow(<ExpensesSummary expenses={expenses} />);
  expect(wrapper).toMatchSnapshot();
});

test('should render ExpensesSummary with single expense item', () => {
  const wrapper = shallow(<ExpensesSummary expenses={[expenses[1]]} />);
  expect(wrapper).toMatchSnapshot();
});

test('should render ExpensesSummary with no expense items', () => {
  const wrapper = shallow(<ExpensesSummary expenses={[]} />);
  expect(wrapper).toMatchSnapshot();
});
