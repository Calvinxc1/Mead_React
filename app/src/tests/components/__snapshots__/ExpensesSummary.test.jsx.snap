// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ExpensesSummary with multiple expense items 1`] = `
<div>
  Viewing 
  3
   expenses totalling 
  $1,141.95
</div>
`;

exports[`should render ExpensesSummary with no expense items 1`] = `
<div>
  Viewing 
  0
   expenses totalling 
  $0.00
</div>
`;

exports[`should render ExpensesSummary with single expense item 1`] = `
<div>
  Viewing 
  1
   expenses totalling 
  $1,095.00
</div>
`;
